File: android/src/main/java/com/facebook/flipper/plugins/inspector/descriptors/utils/stethocopies/FragmentAccessor.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.flipper.plugins.inspector.descriptors.utils.stethocopies;

import android.content.res.Resources;
import android.view.View;
import javax.annotation.Nullable;

public interface FragmentAccessor<FRAGMENT, FRAGMENT_MANAGER> {
  int NO_ID = 0;

  @Nullable
  FRAGMENT_MANAGER getFragmentManager(FRAGMENT fragment);

  Resources getResources(FRAGMENT fragment);

  int getId(FRAGMENT fragment);

  @Nullable
  String getTag(FRAGMENT fragment);

  @Nullable
  View getView(FRAGMENT fragment);

  @Nullable
  FRAGMENT_MANAGER getChildFragmentManager(FRAGMENT fragment);
}

File: android/src/main/java/com/facebook/flipper/plugins/uidebugger/stetho/FragmentAccessor.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.flipper.plugins.uidebugger.stetho

import android.content.res.Resources
import android.view.View

interface FragmentAccessor<FRAGMENT, FRAGMENT_MANAGER> {
  fun getFragmentManager(fragment: FRAGMENT): FRAGMENT_MANAGER?
  fun getResources(fragment: FRAGMENT): Resources?
  fun getId(fragment: FRAGMENT): Int
  fun getTag(fragment: FRAGMENT): String?
  fun getView(fragment: FRAGMENT): View?
  fun getChildFragmentManager(fragment: FRAGMENT): FRAGMENT_MANAGER?

  companion object {
    const val NO_ID = 0
  }
}
